cpp/tests/streams/quantile_test.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/table_utilities.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/column/column_view.hpp>
+#include <cudf/detail/tdigest/tdigest.hpp>
+#include <cudf/quantiles.hpp>
+#include <cudf/table/table_view.hpp>
+#include <cudf/types.hpp>
+
+#include <stdexcept>
+
+struct QuantileTest : public cudf::test::BaseFixture {};
+
+TEST_F(QuantileTest, TestMultiColumnUnsorted)
+{
+  auto input_a = cudf::test::strings_column_wrapper(
+    {"C", "B", "A", "A", "D", "B", "D", "B", "D", "C", "C", "C",
+     "D", "B", "D", "B", "C", "C", "A", "D", "B", "A", "A", "A"},
+    {true, true, true, true, true, true, true, true, true, true, true, true,
+     true, true, true, true, true, true, true, true, true, true, true, true});
+
+  cudf::test::fixed_width_column_wrapper<numeric::decimal32, int32_t> input_b(
+    {4, 3, 5, 0, 1, 0, 4, 1, 5, 3, 0, 5, 2, 4, 3, 2, 1, 2, 3, 0, 5, 1, 4, 2},
+    {1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1});
+
+  auto input = cudf::table_view({input_a, input_b});
+
+  auto actual = cudf::quantiles(input,
+                                {0.0f, 0.5f, 0.7f, 0.25f, 1.0f},
+                                cudf::interpolation::NEAREST,
+                                cudf::sorted::NO,
+                                {cudf::order::ASCENDING, cudf::order::DESCENDING},
+                                {},
+                                cudf::test::get_default_stream());
```

Comment:
I would add some CUDF_TEST_EXPECT_TABLES_EQUAL (or whichever is appropriate) checks to these tests.

Reply:
Stream tests only verify if the stream is being forwarded correctly. We do not check functional correctness here; the test tracks the test stream passed and verifies that it is used correctly for all operations.

Reply:
Naively, this code looks like it's intending to test whether these cudf functions (e.g. quantiles()) work properly.  If we don't want to test that here, please add small comments to that effect.